src/plugins/cordova-plugin-device-orientation/compass.js

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+var db = require('db');
+
+var constants = {
+    HEADING_KEY: 'device-orientation-heading-key'
+};
+
+var compass = {
+    heading: null,
+
+    initialize: function () {
+        var value = db.retrieve(constants.HEADING_KEY);
+
+        this.heading = parseFloat(value) || 0;
+    },
+
+    /**
+     * @param {number} value
+     */
+    updateHeading: function (value) {
+        this.heading = parseFloat(value);
+
+        db.save(constants.HEADING_KEY, this.heading);
+    },
+
+    /**
+     * The exec call to Compass.getHeading executes this function.
+     * @param {function} success The function used to simulate the native call to CallbackContext.sendPluginResult.
+     * @param {function} error The function used to simulate the native call to CallbackContext.error
+     */
+    getHeading: function (success, error) {
+        if (typeof success === 'function') {
+            success(compass.getHeadingResult());
+        }
+    },
+
+    /**
+     * The exec call to Compass.stopHeading executes this function.
+     */
+    stopHeading: function () {
+        // TODO implement for iOS
```

Comment:
After digging a little into the implementation of the cordova-device-orientation plugin, there's no need to have some implementation specific for iOS simulation. The reason of the TODO comment was that for iOS there's no need to create timers in javascript side, comparing for the other platforms. This is identified by a UUID value generated using utils module from cordova. But when running on the browser and simulator, that validation never applies, and the timers are set in place.
Removing this TODO and executing the success callback.